message_ix_models/testing.py

```diff
@@ -207,3 +222,121 @@ def bare_res(request, context: Context, solved: bool = False) -> message_ix.Scen
 
     log.info(f"Clone to '{name}/{new_name}'")
     return base.clone(scenario=new_name, keep_solution=solved)
+
+
+#: Items with names that match (partially or fully) these names are omitted by
+#: :func:`export_test_data`.
+EXPORT_OMIT = [
+    "aeei",
+    "cost_MESSAGE",
+    "demand_MESSAGE",
+    "demand",
+    "depr",
+    "esub",
+    "gdp_calibrate",
+    "grow",
+    "historical_gdp",
+    "kgdp",
+    "kpvs",
+    "lakl",
+    "land",
+    "lotol",
+    "mapping_macro_sector",
+    "MERtoPPP",
+    "prfconst",
+    "price_MESSAGE",
+    "ref_",
+    "sector",
+]
+
+
+def export_test_data(context: Context):
+    """Export a subset of data from a scenario, for use in tests.
+
+    The context settings ``export_nodes`` (default: "R11_AFR" and "R11_CPA") and
+    ``export_techs`` (default: "coal_ppl") are used to filter the data exported.
+    In addition, any item (set, parameter, variable, or equation) with a name matching
+    :data:`EXPORT_OMIT` *or* the context setting ``export_exclude`` is discarded.
+
+    The output is stored at :file:`data/tests/{model name}_{scenario name}_{techs}.xlsx`
+    in :mod:`message_data`.
+
+    See also
+    --------
+    :ref:`export-test-data`
+    """
+    from message_ix_models.util import private_data_path
+
+    # Load the scenario to be exported
+    scen = context.get_scenario()
+
+    # Retrieve the context settings giving the nodes and technologies to export
+    nodes = context.get("export_nodes", ["R11_AFR", "R11_CPA"])
+    technology = context.get("export_techs", ["coal_ppl"])
+
+    # Construct the destination file name
+    dest_file = private_data_path(
```

Comment:
Further testing of the command showed that a useful feature may be to allow a suffix to be added to the file output. This avoids files being overwritten accidentally.

Reply:
I think this is somewhat overkill. This is a tool for development. It's on the individual developer to do things like:

- think about the test for which data is being exported,
- use appropriate options,
- choose names and locations that make clear what the data is and how it related to certain tests,
- handle files in their git working directory.

The code can't possibly anticipate all that, so I think it's best if it doesn't pretend to try.